opentelemetry-auto-instrumentation/README.md

```diff
@@ -0,0 +1,151 @@
+```
+
+Instrument Rails application with `bundle exec`:
+
+Since installing the `opentelemetry-auto-instrumentation` gem through `gem install`, anything related to the OpenTelemetry gem won't be stored in Bundler's gem path. Therefore, users need to add an additional gem path that contains these gems prior to initialization.
+
+```console
+RUBYOPT="-r {PUT YOUR GEM PATH}/gems/opentelemetry-auto-instrumentation-0.1.0/lib/opentelemetry-auto-instrumentation" bundle exec rails server
+```
+
+Instrument Sinatra application with rackup:
+
+If you are using a Gemfile to install the required gems but without `Bundler.require`, set `REQUIRE_BUNDLER` to true. This way, `opentelemetry-auto-instrumentation` will call `Bundler.require` to initialize the required gems prior to SDK initialization.
+
+```console
+export REQUIRE_BUNDLER=true
+RUBYOPT="-r opentelemetry-auto-instrumentation" rackup config.ru
+```
+
+If you wish to load some gems outside the Gemfile, then they need to be placed in front of opentelemetry-auto-instrumentation:
+
+```console
+export BUNDLE_WITHOUT=development,test
+gem install mysql2
+RUBYOPT="-r mysql2 -r opentelemetry-auto-instrumentation" ruby application.rb
+```
+
+## Example
+
+In example folder, executing the following commands should result in trace output.
+
+```console
+# if the user doesn't want to install opentelemetry-auto-instrumentation from rubygems.org
+# the user can build the gem and install it with gem install *.gem
+
+gem install opentelemetry-auto-instrumentation
+bundle install
+export REQUIRE_BUNDLER=true
+export OTEL_TRACES_EXPORTER=console
+RUBYOPT="-r opentelemetry-auto-instrumentation" ruby app.rb
+```
+
+## Configuration
+
+These environment variables are not standard OpenTelemetry environment variables; they are only feature flags for this gem.
+
+| Environment Variable | Description | Default | Example |
+|----------------------|-------------|---------|---------|
+| `OTEL_RUBY_REQUIRE_BUNDLER` | Set to `true` if you are using Bundler to install gems but without `Bundler.require` in your script during initialization. | nil | N/A |
+| `OTEL_RUBY_ADDITIONAL_GEM_PATH` | Intended to be used for the OpenTelemetry Operator environment if you install `opentelemetry-auto-instrumentation` to a customized path. | nil | N/A |
+| `OTEL_RUBY_OPERATOR` | Set to `true` to set the binding path for the OpenTelemetry Operator. | `/otel-auto-instrumentation-ruby` | N/A |
+| `OTEL_RUBY_RESOURCE_DETECTORS` | Determine what kind of resource detector is needed. Currently supports `container`, `azure`, and `google_cloud_platform`. Use commas to separate multiple detectors. | nil | `container,azure` |
+| `OTEL_RUBY_ENABLED_INSTRUMENTATIONS` | Shortcut used when you only want to instrument one or a couple of particular libraries. | nil | `redis,active_record` |
+| `OTEL_RUBY_ZERO_CODE_DEBUG` | Set to `true` if want to see some debug information. This to avoid preload logger gem | nil | N/A |
```

Comment:
I think we should rename this to `OTEL_RUBY_AUTO_INSTRUMENTATION_DEBUG` since we're not using zero code as the name anymore.